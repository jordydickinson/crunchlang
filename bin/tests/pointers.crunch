fun main!() {
    var x = 1;
    var y: int32* = &x;
    *y := 2;
}
