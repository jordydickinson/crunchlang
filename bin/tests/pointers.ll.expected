; ModuleID = 'pointers.crunch'
source_filename = "pointers.crunch"

@llvm.global_ctors = appending global [1 x void ()*] [void ()* @init.ctors]

define void @main() {
entry:
  br label %body

body:                                             ; preds = %entry
  %x = alloca i32
  store i32 1, i32* %x
  %y = alloca i32*
  store i32* %x, i32** %y
  %y.deref = load i32*, i32** %y
  store i32 2, i32* %y.deref
  ret void
}

define void @init.ctors() {
entry:
  ret void
}
